patches/@react-navigation+stack+6.3.16.patch

```diff
@@ -0,0 +1,58 @@
+diff --git a/node_modules/@react-navigation/stack/src/views/Stack/StackView.tsx b/node_modules/@react-navigation/stack/src/views/Stack/StackView.tsx
+index 6bbce10..5139382 100644
+--- a/node_modules/@react-navigation/stack/src/views/Stack/StackView.tsx
++++ b/node_modules/@react-navigation/stack/src/views/Stack/StackView.tsx
+@@ -375,29 +375,48 @@ export default class StackView extends React.Component<Props, State> {
+ 
+   private handleTransitionStart = (
+     { route }: { route: Route<string> },
+-    closing: boolean
+-  ) =>
++    closing: boolean,
++  ) => {
+     this.props.navigation.emit({
+       type: 'transitionStart',
+       data: { closing },
+       target: route.key,
+     });
++  }
+ 
+   private handleTransitionEnd = (
+     { route }: { route: Route<string> },
+-    closing: boolean
+-  ) =>
++    closing: boolean,
++    state: StackNavigationState<ParamListBase>
++  ) => {
+     this.props.navigation.emit({
+       type: 'transitionEnd',
+       data: { closing },
+       target: route.key,
+     });
++    if (state?.index > 1) {
++      this.props.navigation.emit({
++        type: 'transitionEnd',
++        data: { closing: !closing },
++        target: state.routes[state.index - 2].key,
++      });
++    }
```

Comment:
It is not really a bug that would be fixed in `react-navigation`, but it has to be introduced specifically for the architecture of nested navigators and usage of `Freeze` component in the Expensify app. In here we just make sure that the second to last screen in the stack gets information that it cannot be seen. It is necessary, because otherwise a frozen screen could be seen below ie. while using the swipe gesture on iOS.